### DISCUSSION TITLE
Logical or?

### DISCUSSION THREAD

**Discussion by deklanw** [JuliaSIMD/LoopVectorization.jl#328] [Q&A] [answered]
Trying to `turbo` this loop

```julia
nll_delta = 0

@turbo for b in 1:B
    eq = b==r || b==s
    t1 = -E[r,b] * log(E[r,b] / (E_s_r * E_sums[b]))
    t1 = ifelse(E[r, b]==0, 0.0, ifelse(eq, t1/2, t1))
    t2 = -E[s,b] * log(E[s,b] / (E_s_s * E_sums[b]))
    t2 = ifelse(E[s, b]==0, 0.0, ifelse(eq, t2/2, t2))
    
    nll_delta += t1 + t2
end
```

Getting
```
LoadError: Expression not recognized.
b == r || b == s
```

I'm thinking I can nest even more `ifelse` statements but it's already getting unreadable. Is there another way?

**Comment by chriselrod** [comment, marked as answer]
```julia
eq = (b == r) | (b == s)
```
Note that the parenthesis are needed, because `|` has difference precedence from `||`.

**Reply by deklanw** [reply]
ahhh I was wondering why I was getting non-sense when I left off the parentheses. Thanks

**Reply by chriselrod** [reply]
Yeah.
```julia
julia> 31 == 7 | 24 == 31
true
```
Precedence table:
https://docs.julialang.org/en/v1/manual/mathematical-operations/#Operator-Precedence-and-Associativity
Many logical operators are grouped with addition.
 Seems `&` is with `*`, though.
